llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1693,6 +1693,15 @@ InstructionCost VPWidenRecipe::computeCost(ElementCount VF,
     if (RHS->isLiveIn())
       RHSInfo = Ctx.TTI.getOperandInfo(RHS->getLiveInIRValue());
 
+    // The mul is folded into another target instruction when participating
+    // in scaled reductions.
+    if (Opcode == Instruction::Mul && !hasMoreThanOneUniqueUser()) {
+      if (all_of(users(), [](const VPUser *U) {
+            return isa_and_present<VPPartialReductionRecipe>(U);
+          }))
+        return TTI::TCC_Free;
```

Comment:
See my comment on `isScaledReductionExtension` - I have concerns this is essentially AArch64-specific code being used for all targets.